File: Functions/FinancialCalcUpdate.py
# Eve Richardson Financial Calculator Update Python

def info(income, amount, type):
    typepc = amount/income*100
    print(f"You spend ${amount:.2f} on {type} and that is {typepc:.2}% of your income.")

def ask(type):
    return input(f"What is your monthly {type}:\n")

print("Hello, welcome to my budget calculator.\n")

income = float(ask("income"))
rent = float(ask("rent"))
utilities = float(ask("utilities spending"))
groceries = float(ask("groceries spending"))
transpo = float(ask("transportation spending"))

savings = float(income/10)
spending = float(income-(rent+utilities+transpo+groceries+savings))
spendingpc = float(spending/income)*100

info(income, rent, "rent")
info(income, utilities, "utilities")
info(income, groceries, "groceries")
info(income, transpo, "transpo")
print(f"You should save ${savings:.2f} and that is 10% of your income.")
print(f"You have ${spending:.2f} left to spend and that is {spendingpc:.2f}% of your income.")

File: Functions/FinancialCalcUpdate.c
// Eve Richardson, Financial Calculator Update C
#include <stdio.h>


float income, rent, utilities, grocery, transpo, spend, save, total;




float ask(char type[], float money){
   printf("What is your monthly %s:\n", type);
   scanf("%f", &money);
   return money;
}




void percents(char type[], int cost){
   int per = cost/income*100;
   printf("Your %s is %d%% of your income.\n", type, per);
  
}




int main(void){
  printf("This is a budget calculator.\n");
  income = ask("income", income);
  rent = ask("rent", rent);
  transpo = ask("transportation", transpo);
  utilities = ask("utilities", utilities);
  grocery = ask("groceries", grocery);


  save = income/10;
  total = rent+utilities+grocery+transpo;
  spend = income-total-save;


  printf("You make $%.2f\n", income);
  printf("You have spent $%.2f total.\n", total);
  printf("You have $%.2f left to spend.\n", spend);


  percents("rent", rent);
  percents("groceries", grocery);
  percents("utilities", utilities);
  percents("transportation", transpo);
  percents("leftover spending money", spend);
  printf("You should save 10 percent of your income and that is $%.2f\n", save);
 
   return 0;
}



